src/services/completions.ts

```diff
@@ -2086,8 +2086,7 @@ namespace ts.Completions {
                 // GH#39946. Pulling on the type of a node inside of a function with a contextual `this` parameter can result in a circularity
                 // if the `node` is part of the exprssion of a `yield` or `return`. This circularity doesn't exist at compile time because
                 // we will check (and cache) the type of `this` *before* checking the type of the node.
-                const container = getThisContainer(node, /*includeArrowFunctions*/ false);
-                if (!isSourceFile(container) && container.parent) typeChecker.getTypeAtLocation(container);
+                typeChecker.tryGetThisTypeAt(node, /*includeGlobalThis*/ false);

```

Comment:
Why is calling `tryGetThisTypeAt` better than calling `getTypeAtLocation`?

Reply:
Please disregard, I took a closer look and have a better understanding of the change here.

Reply:
@rbuckton Thanks for the review. Sorry for the confusion, I had to add notes on why `tryGetThisTypeAt` was used.